CDP4Composition/Views/LoadingControl.xaml.cs

```diff
@@ -1,24 +1,88 @@
-﻿// -------------------------------------------------------------------------------------------------
+﻿// --------------------------------------------------------------------------------------------------------------------
 // <copyright file="LoadingControl.cs" company="RHEA System S.A.">
-//   Copyright (c) 2015 RHEA System S.A.
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Merlin Bieze, Naron Phou, Patxi Ozkoidi, Alexander van Delft, Mihail Militaru
+//            Nathanael Smiechowski, Kamil Wojnowski
+//
+//    This file is part of CDP4-IME Community Edition. 
+//    The CDP4-IME Community Edition is the RHEA Concurrent Design Desktop Application and Excel Integration
+//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 // </copyright>
-// -------------------------------------------------------------------------------------------------
+// --------------------------------------------------------------------------------------------------------------------
 
 namespace CDP4Composition.Views
 {
-    using System;
     using System.Windows;
     using System.Windows.Controls;
-    using DevExpress.Xpf.Grid;
+    using System.Windows.Input;
 
     /// <summary>
     /// Interaction logic for LoadingControl.xaml
     /// </summary>
     public partial class LoadingControl : UserControl
     {
+        /// <summary>
+        /// The dependency property that allows setting the cancel <see cref="ICommand"/>
+        /// </summary>
+        public static readonly DependencyProperty CancelCommandProperty = DependencyProperty.Register("CancelCommand", typeof(ICommand), typeof(LoadingControl), new FrameworkPropertyMetadata(null));
```

Comment:
I dont think you need to do this. If the user control does not have a datacontext then it should automatically bind to properties of the viewmodel of its container. So it directly binds. In fact I think it already does this. In the xaml you use `Command="{Binding...` which already points to the data context. So if you remove there `ElementName` mumbo jumbo it should in theory work.

Reply:
The usage XAML files of the LoadingControl only know the LoadingControl.

Reply:
The binding is weak, it knows at runtime when the data context is assigned. I will bet 5 bucks on it if you do a screenshare :)

Reply:
It will indeed work, but then you have an "undocumented API" there.
With this implementation you see a binding in the XAML file where you'd expect it.

Reply:
Well that's another point altogether, and sounds valid enough :)